File: 1046-max-consecutive-ones-iii/1046-max-consecutive-ones-iii.cpp
class Solution {
public:
    int longestOnes(vector<int>& nums, int k) {
        int l = 0, r = 0, count = 0, n = nums.size();
        while(r<n && count<=k){
            if(nums[r] == 0) count++;
            r++;
        }
        if(count<=k) return n;
        int maxi = r - l - 1;
        while(r<n){
            if(count > k){
                while(count>k){
                    if(nums[l] == 0) count--;
                    l++;
                }
                maxi = max(maxi, r - l);
            }
            else{
                while(r<n && count<=k){
                    if(nums[r] == 0) count++;
                    r++;
                }
                cout<<r<<" "<<l<<endl;
                maxi = max(maxi, r - l - 1);
            }
        }
        if(count <= k) maxi = max(maxi, r - l);
        return maxi;
    }
};

File: 1046-max-consecutive-ones-iii/1046-max-consecutive-ones-iii.go
func longestOnes(nums []int, k int) int {
    n := len(nums)
    l, r := 0, -1
    z, ans := 0, 0
    for r < n {
        for z > k {
            if nums[l] == 0 {
                z--
            }
            l++
        }
        for r < n && z < k+1 {
            r++
            if r == n {
                break
            }
            if nums[r] == 0 {
                z++
            }
        }
        ans = max(r - l, ans)
    }
    return ans
}